### ISSUE TITLE
Allow random extensions

### ISSUE THREAD

**Issue by chevdor** [chevdor/tera-cli#17]
We may want to pass a custom extension such as `.foobar`.
In that case, we could pass an extra flag to provide the format (such as `--yaml`) or be smart and guess.

**Comment by alerque** [comment]
This would also be useful for being able handle process substitution without so much song and dance to get extensions sorted:

```zsh
TMPSUFFIX=.yml
tera -t template <(command_that_generates_context)
unset TMPSUFFIX
```

Could be:

```zsh
tera -t template --yaml <(command_that_generates_context)
```

Personally I would suggest *not* trying to be smart and guess the type if it isn't noted as an extension. YAML is a disaster to detect because it's technically a superset of JSON, so most JSON is valid YAML and YAML can be made to look very much like JSON but actually work the same way a JSON parser would handle it.

**Comment by chevdor** [comment]
I agree, I am only using the detection based on the extension so spare the user yet another flag but the current rules are rather tight.
